config/redis.go

```diff
@@ -0,0 +1,50 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/0xERR0R/blocky/log"
+)
+
+// RedisConfig configuration for the redis connection
+type RedisConfig struct {
+	Addresses          []string `yaml:"addresses"`
+	Username           string   `yaml:"username" default:""`
+	Password           string   `yaml:"password" default:""`
+	SentinelUsername   string   `yaml:"sentinelUsername" default:""`
+	SentinelPassword   string   `yaml:"sentinelPassword" default:""`
+	SentinelMasterSet  string   `yaml:"sentinelMasterSet" default:""`
+	Database           int      `yaml:"database" default:"0"`
+	ConnectionAttempts int      `yaml:"connectionAttempts" default:"3"`
+	ConnectionCooldown Duration `yaml:"connectionCooldown" default:"1s"`
+	ConnRingScale      int      `yaml:"connRingScale" default:"10"`
+	ShuffleInit        bool     `yaml:"shuffleInit" default:"false"`
+	LocalCacheTime     Duration `yaml:"localCacheTime" default:"10m"`
+	LocalCacheSize     int      `yaml:"localCacheSize" default:"1048576"`
+	Required           bool     `yaml:"required" default:"false"` // Deprecated: always required if enabled
+	Address            string   `yaml:"address"`                  // Deprecated: use Addresses
+	SentinelAddresses  []string `yaml:"sentinelAddresses"`        // Deprecated: use Addresses
+}
+
+func fixDeprecatedRedis(cfg *Config) {
+	if len(cfg.Redis.Addresses) == 0 && len(cfg.Redis.SentinelAddresses) > 0 {
+		log.Log().Warnln("'redis.sentinelAddresses' is deprecated. Please use 'redis.addresses' instead.")
+
+		cfg.Redis.Addresses = cfg.Redis.SentinelAddresses
+	} else if len(cfg.Redis.Addresses) == 0 && len(cfg.Redis.Address) > 0 {
```

Comment:
Should the `else` be removed here so we use both `SentinelAddresses` and `Address` if they're both set?  
Might be obvious we don't want that but I don't know Redis enough :smile: 

The `len(cfg.Redis.Addresses) == 0` condition would need to be evaluated before the first if to still work.

Reply:
Nope rueidis uses only one address pool which only contains the sentinel addresses if sentinel is used.
Otherwise the pool contains all redis instances(multiple entries if a redis cluster is used).
A mix of sentinel and redis addresses is not allowed (as far as I know).